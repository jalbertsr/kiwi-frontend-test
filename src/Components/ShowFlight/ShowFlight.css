.flightContainer {
  margin-top: 4%;
}
