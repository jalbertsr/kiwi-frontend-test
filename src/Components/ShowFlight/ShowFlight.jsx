import React, { Component } from 'react'
import { Row, Col } from 'antd'
import PropTypes from 'prop-types'
import FlightDescription from '../FlightDescription/FlightDescription'

import Service from '../../Services/flightService'
import FlightMap from '../Map/Map'
import styles from './ShowFlight.css'

export default class ShowFlight extends Component {
  constructor (props) {
    super(props)
    this.state = {
      result: {}
    }
  }

  componentDidMount () {
    this._componentDidMount()
  }

  async _componentDidMount () {
    if (this.props.match.params.query) {
      const [flyFrom, flyTo, departure, returning, index] =
        this.props.match.params.query.split('&').map(query => query.split('=')[1])
      const parsedDeparture = departure.replace('-', '/').replace('-', '/')
      const parsedReturn = returning.replace('-', '/').replace('-', '/')

      const data = await Service.getFlights(flyFrom, flyTo, parsedDeparture, parsedReturn)
      try {
        this.setState({
          result: data.data[index]
        })
      } catch (e) {
        throw new Error(e)
      }
    }
  }

  render () {
    const result = this.state.result
    return (
      <div className={styles.flightContainer}>
        <Row>
          <Col xs={18} md={8} offset={1}>
            { result.route &&
              <FlightMap
                latFrom={result.route['0'].latFrom}
                latTo={result.route['0'].latTo}
                lngFrom={result.route['0'].lngFrom}
                lngTo={result.route['0'].lngTo}
              />
            }
          </Col>
          <Col xs={18} md={12} offset={1}>
            {
              result.route &&
              <FlightDescription {...result} />
            }
          </Col>
        </Row>
      </div>
    )
  }
}

ShowFlight.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      query: PropTypes.string.isRequired
    })
  })
}
