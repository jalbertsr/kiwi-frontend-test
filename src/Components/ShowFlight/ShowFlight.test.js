import ShowFlight from './ShowFlight'
import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import { shallow, configure } from 'enzyme'

configure({ adapter: new Adapter() })

/* eslint-disable no-unused-vars */

describe('ShowFlight', () => {
  const props = {
    match: {
      params: {
        query: 'flight/flyFrom=BCN&flyTo=BRQ&departure=01-11-2017&returning=06-11-2017'
      }
    }
  }
  test('renders correctly', () => {
    const component = shallow(<ShowFlight {...props} />)
    expect(component).toMatchSnapshot()
  })
})
