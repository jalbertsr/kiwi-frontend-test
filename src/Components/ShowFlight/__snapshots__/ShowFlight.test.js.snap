// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowFlight renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ShowFlight
    match={
      Object {
        "params": Object {
          "query": "flight/flyFrom=BCN&flyTo=BRQ&departure=01-11-2017&returning=06-11-2017",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Row
        gutter={0}
      >
        <Col
          md={8}
          offset={1}
          xs={18}
        />
        <Col
          md={12}
          offset={1}
          xs={18}
        />
      </Row>,
      "className": "flightContainer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Col
            md={8}
            offset={1}
            xs={18}
          />,
          <Col
            md={12}
            offset={1}
            xs={18}
          />,
        ],
        "gutter": 0,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
            "md": 8,
            "offset": 1,
            "xs": 18,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
            "md": 12,
            "offset": 1,
            "xs": 18,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Row
          gutter={0}
        >
          <Col
            md={8}
            offset={1}
            xs={18}
          />
          <Col
            md={12}
            offset={1}
            xs={18}
          />
        </Row>,
        "className": "flightContainer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Col
              md={8}
              offset={1}
              xs={18}
            />,
            <Col
              md={12}
              offset={1}
              xs={18}
            />,
          ],
          "gutter": 0,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": undefined,
              "md": 8,
              "offset": 1,
              "xs": 18,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": undefined,
              "md": 12,
              "offset": 1,
              "xs": 18,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
